vendor/engines/badges_engine/app/controllers/badges_engine/admin/badges_controller.rb

```diff
@@ -0,0 +1,72 @@
+require_dependency 'badges_engine/application_controller'
+
+module BadgesEngine
+  class Admin::BadgesController < Admin::BaseController
+    before_action :set_badge, only: [:edit, :update, :destroy]
+    helper_method :sort_column, :sort_direction
+    layout 'admin'
+
+    def index
+      @badges = if params[:search]
+                  Badge.where("name ILIKE ?", "%#{params[:search]}%").order(sort_column + ' ' + sort_direction)
```

Comment:
check your spacing, I'm not completely sure but I guess the convention here at Crowd is to indent just 4 spaces of the previous line...